### ISSUE TITLE
Duplicated notifications are possibly created in Cypress

### ISSUE THREAD

**Issue by Silver-IT** [okTurtles/group-income#2230]
### Problem

In [this Cypress test](https://cloud.cypress.io/projects/q6whky/runs/2743/test-results/5c8cbd1f-a8e6-489a-8b31-b6b7e8e379d8/video?actions=%5B%5D&browsers=%5B%5D&groups=%5B%5D&isFlaky=%5B%5D&modificationDateRange=%7B%22startDate%22%3A%221970-01-01%22%2C%22endDate%22%3A%222038-01-19%22%7D&orderBy=EXECUTION_ORDER&oses=%5B%5D&specs=%5B%5D&statuses=%5B%7B%22value%22%3A%22FAILED%22%2C%22label%22%3A%22FAILED%22%7D%5D&testingTypesEnum=%5B%5D&utm_source=github), we can see the two notifications are same.
![image](https://github.com/user-attachments/assets/ee68d124-5541-4522-8972-9ff2d2ed60ec)

Actually, it is not allowed to create duplicated notification and we do this by comparing its hash.
```
[keys.ADD_NOTIFICATION] (state, notification: Notification) {
  if (state.items.some(item => item.hash === notification.hash)) {
    // We cannot throw here, as this code might be called from within a contract side effect.
    return console.error('This notification is already in the store.')
  }
  state.items.push(notification)
  // Sort items in chronological order, newest items first.
  state.items.sort(compareOnTimestamp)
}
```
I guess the reason might be the duplicated notification is created before the Vuex module `notifications` is cloned from the IndexedDB after login.

### Solution

Fix it.

**Comment by Silver-IT** [comment]
Or the reason could be the `identityContractID` which is inside the data could be different.

**Comment by Silver-IT** [comment]
Having investigated more in detail, I have noticed that those two notifications are different ones. Just difference in period.
In my test the difference in period is just like the following.
```
The first one:  "2024-07-19T06:58:48.784Z"
The second one: "2024-07-19T06:57:26.630Z"
```

Also, I have noticed that it's related to the force change of `distributionDate` which is updated by `forceDistributionDate` function inside `cy.giForceDistributionDateToNow()`.
